gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java

```diff
@@ -397,24 +445,28 @@ private void finalizeDatasetStateBeforeCommit(JobState.DatasetState datasetState
   private boolean canCommitDataset(JobState.DatasetState datasetState) {
     // Only commit a dataset if 1) COMMIT_ON_PARTIAL_SUCCESS is used, or 2)
     // COMMIT_ON_FULL_SUCCESS is used and all of the tasks of the dataset have succeeded.
-    return this.jobCommitPolicy == JobCommitPolicy.COMMIT_ON_PARTIAL_SUCCESS ||
-        this.jobCommitPolicy == JobCommitPolicy.COMMIT_SUCCESSFUL_TASKS ||
-        (this.jobCommitPolicy == JobCommitPolicy.COMMIT_ON_FULL_SUCCESS &&
-            datasetState.getState() == JobState.RunningState.SUCCESSFUL);
+    return this.jobCommitPolicy == JobCommitPolicy.COMMIT_ON_PARTIAL_SUCCESS
+        || this.jobCommitPolicy == JobCommitPolicy.COMMIT_SUCCESSFUL_TASKS
+        || (this.jobCommitPolicy == JobCommitPolicy.COMMIT_ON_FULL_SUCCESS
+            && datasetState.getState() == JobState.RunningState.SUCCESSFUL);
   }
 
   /**
    * Commit the output data of a dataset.
    */
   @SuppressWarnings("unchecked")
-  private void commitDataset(JobState.DatasetState datasetState) throws IOException {
+  private Optional<CommitSequence.Builder> commitDatasetOrGenerateCommitSequenceBuilder(
```

Comment:
Why not making `Optional<CommitSequence.Builder>` a class variable, and keep the return type `void`?